### DISCUSSION TITLE
Proposal: Replacing aiohttp with FastAPI in ComfyUI Backend

### DISCUSSION THREAD

**Discussion by bigcat88** [Comfy-Org/ComfyUI#5931] [Ideas]
Would you be open to accepting a pull request that replaces `aiohttp` with FastAPI?

---

We recently completed implementing our HTTP/1.1 API in Visionatrix to handle tasks for ComfyUI, but we still need to retain the original ComfyUI frontend interface.

Since we're using FastAPI, it's not straightforward to integrate ComfyUI's endpoints into our project without reimplementing them from scratch.

While I'm confident I could reimplement the endpoints myself, and they would likely maintain full backward compatibility with the original `aiohttp` endpoints, it would be far more convenient to avoid doing this in our own repository. Constantly syncing backend changes from ComfyUI would be cumbersome. Instead, it would be ideal to submit a PR to this repository, replacing `aiohttp` with FastAPI.

With that change, we could simply import the routes into our project, making everything seamless and easy to manage.

So, my question for you, @comfyanonymous, @mcmonkey4eva, is: would you consider accepting a PR that replaces `aiohttp` with FastAPI if it will retain backward compatibility with current endpoints?

Thank you in advance for your consideration, no matter the response!

**Comment by catboxanon** [comment]
I think you may want to open an issue in the https://github.com/Comfy-Org/rfcs repo to get more feedback on this.
Maybe do the same for https://github.com/Comfy-Org/ComfyUI_frontend/discussions/1806 too.

**Reply by bigcat88** [reply]
Certainly, it's possible to explore and discuss this. However, due to the difficulty in getting timely feedback from the ComfyUI developers (as seen in PR #6376), our team at Nextcloud—where we were considering offering ComfyUI to the community—has currently put that plan on hold until the situation improves.
